dags/datasets/police-annual-statistical-report.py

```diff
@@ -0,0 +1,223 @@
+# police-annual-statistical-report.py - this file makes multiple DAGs; each is an ETL for a police annual report from AGOL into CKAN
+
+import ckanapi
+import json
+
+from pathlib import Path
+from datetime import datetime
+
+from airflow import DAG
+from airflow.models import Variable
+from airflow.operators.dummy import DummyOperator
+from airflow.operators.python import BranchPythonOperator, PythonOperator
+from airflow.operators.slack_operator import SlackAPIPostOperator
+
+from utils_operators.directory_operator import CreateLocalDirectoryOperator, DeleteLocalDirectoryOperator
+from utils_operators.agol_operators import AGOLDownloadFileOperator
+from utils_operators.slack_operators import task_success_slack_alert, task_failure_slack_alert, GenericSlackOperator
+
+from ckan_operators.datastore_operator import (
+    BackupDatastoreResourceOperator,
+    DeleteDatastoreResourceRecordsOperator,
+    InsertDatastoreResourceRecordsOperator,
+    RestoreDatastoreResourceBackupOperator,
+    DeleteDatastoreResourceOperator,
+    InsertDatastoreResourceRecordsFromJSONOperator
+)
+from ckan_operators.resource_operator import GetOrCreateResourceOperator, EditResourceMetadataOperator
+from utils import airflow_utils
+
+
+
+# init hardcoded vars for these dags
+ACTIVE_ENV = Variable.get("active_env")
+CKAN_CREDS = Variable.get("ckan_credentials_secret", deserialize_json=True)
+CKAN = CKAN_CREDS[ACTIVE_ENV]["address"]
+CKAN_APIKEY = CKAN_CREDS[ACTIVE_ENV]["apikey"]
+
+TMP_DIR = Path(Variable.get("tmp_dir"))
+
+base_url = "https://services.arcgis.com/S9th0jAJ7bqgIRjw/arcgis/rest/services/"
+
+datasets = [
+    {"schema_change": False, "name": "Reported Crimes", "tps_table_code": "ASR-RC-TBL-001", "package_id": "police-annual-statistical-report-reported-crimes", "agol_dataset": "Reported_Crimes_ASR_RC_TBL_001"},
+    {"schema_change": False, "name": "Victims of Crime", "tps_table_code": "ASR-VC-TBL-001", "package_id": "police-annual-statistical-report-victims-of-crime", "agol_dataset": "Victims_of_Crime_ASR_VC_TBL_001"},
+    {"schema_change": False, "name": "Search of Persons", "tps_table_code": "ASR-SP-TBL-001", "package_id": "police-annual-statistical-report-search-of-persons", "agol_dataset": "Search_of_Persons_ASR_SP_TBL_001"},
+    {"schema_change": True, "name": "Traffic Collisions", "tps_table_code": "ASR-T-TBL-001", "package_id": "police-annual-statistical-report-traffic-collisions", "agol_dataset": "Traffic_Collisions_ASR_T_TBL_001"},
```

Comment:
Yeah its intentional - it's the only dataset in the list that had a significant schema change. 

You're right in saying that after the first successful run, we'd have to change this setting to False. Since this dataset is annual (and can change each year) I'm not too concerned.

Reply:
Ah, I see. I expected it set to `False` for consistency with the others but you know best the situation - if there's reason to think this one may have a lot of changes later, may make sense. Quite agree the annual refresh rate means lower priority.